src/main/java/baritone/utils/ToolSet.java

```diff
@@ -90,30 +90,39 @@ private int getMaterialCost(ItemStack itemStack) {
         }
     }
 
+    public boolean hasSilkTouch(ItemStack stack) {
+        return EnchantmentHelper.getEnchantmentLevel(Enchantments.SILK_TOUCH, stack) > 0;
+    }
+
     /**
      * Calculate which tool on the hotbar is best for mining
      *
      * @param b the blockstate to be mined
      * @return A byte containing the index in the tools array that worked best
      */
-    public byte getBestSlot(Block b) {
+    public byte getBestSlot(Block b, boolean preferSilkTouch) {
         byte best = 0;
-        double value = Double.NEGATIVE_INFINITY;
-        int materialCost = Integer.MIN_VALUE;
+        double highestSpeed = Double.NEGATIVE_INFINITY;
+        int lowestCost = Integer.MIN_VALUE;
+        boolean bestSilkTouch = false;
         IBlockState blockState = b.getDefaultState();
         for (byte i = 0; i < 9; i++) {
             ItemStack itemStack = player.inventory.getStackInSlot(i);
-            double v = calculateSpeedVsBlock(itemStack, blockState);
-            if (v > value) {
-                value = v;
+            double speed = calculateSpeedVsBlock(itemStack, blockState);
+            boolean silkTouch = hasSilkTouch(itemStack);
+            if (speed > highestSpeed) {
+                highestSpeed = speed;
                 best = i;
-                materialCost = getMaterialCost(itemStack);
-            } else if (v == value) {
-                int c = getMaterialCost(itemStack);
-                if (c < materialCost) {
-                    value = v;
+                lowestCost = getMaterialCost(itemStack);
+                bestSilkTouch = silkTouch;
+            } else if (speed == highestSpeed) {
+                int cost = getMaterialCost(itemStack);
+                if (cost < lowestCost && (silkTouch || !bestSilkTouch) ||
```

Comment:
Can you add one extra parentheses around the first clause of this plz? While you and I may realize immediately that `a && b || c` becomes `(a && b) || c` it's not always apparent to readers =)